### PR TITLE
Upgrade to Quarkus 3.15.2

### LINKED ISSUE #33469: Remove workaround for overridden micrometer dependencies
### Description

Remove workaround related to overridden Micrometer dependencies introduced in:  https://github.com/keycloak/keycloak/issues/33468

The workaround should be removed after Quarkus including the Micrometer >=1.13.5 version.  

The added code in `keycloak/quarkus/pom.xml`:
```
    <!-- Remove once Quarkus Micrometer extension uses Micrometer 1.13.5-->
    <!-- See https://github.com/keycloak/keycloak/issues/33469 -->
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>io.micrometer</groupId>
                <artifactId>micrometer-core</artifactId>
                <version>1.13.5</version>
            </dependency>
            <dependency>
                <groupId>io.micrometer</groupId>
                <artifactId>micrometer-commons</artifactId>
                <version>1.13.5</version>
            </dependency>
            <dependency>
                <groupId>io.micrometer</groupId>
                <artifactId>micrometer-observation</artifactId>
                <version>1.13.5</version>
            </dependency>
            <dependency>
                <groupId>io.micrometer</groupId>
                <artifactId>micrometer-registry-prometheus-simpleclient</artifactId>
                <version>1.13.5</version>
            </dependency>
        </dependencies>
    </dependencyManagement>
    <!-- Remove until here  -->
```

### LINKED ISSUE #35077: Upgrade to Quarkus 3.15.2
### Description

We should upgrade to the latest LTS patch version of Quarkus. Should be backported to 26.0.x.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mabartos**
- Closes #35077
- Closes #33469

(cherry-picked from commit 02d49b07e8d7efcd2aaef5e4563b872b601497f1)

### CODE REVIEW

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@mabartos Thank you for the PR. Since this is targeting 26 I believe we need to wait with the merge for RHBQ 3.15.2 to be released. @drichtarik Is that correct?

**Comment by mabartos** [comment]
@vmuzikar Thanks for pointing it out. Added `hold` status to wait for the response from @drichtarik.

**Comment by mabartos** [comment]
I'll open another PR once it's merged in main.

**Pull request closed without merging**
